apps/backoffice/src/components/services/subscriptions-migration/migration-manager.tsx

```diff
@@ -0,0 +1,205 @@
+  const [migrationStatusList, setMigrationStatusList] = useState<
+    Array<DelegateStatusPair>
+  >();
+
+  const {
+    data: migrationItemsData,
+    loading: migrationItemsLoading,
+    fetchData: migrationItemsFetchData
+  } = useFetch<MigrationItemList>();
+  const {
+    data: migrationDelegatesData,
+    loading: migrationDelegatesLoading,
+    fetchData: migrationDelegatesFetchData
+  } = useFetch<MigrationDelegateList>();
+
+  const handleOpenImportModal = () => {
+    setIsImportModalOpen(true);
+  };
+
+  const getMigrationOwnershipClaimsUrl = (delegateId: string) =>
+    `${
+      getConfiguration().API_BACKEND_BASE_URL
+    }/api/services/migrations/ownership-claims/${delegateId}`;
+
+  // direct axios fetch
+  const getDelegateMigrationStatus = async (delegateId: string) => {
+    try {
+      const { data, status } = await axios.get<MigrationData>(
+        getMigrationOwnershipClaimsUrl(delegateId)
+      );
+      return { data, status };
+    } catch (error) {
+      if (axios.isAxiosError(error)) {
+        return { data: error.message, status: -1 };
+      } else {
+        return { data: JSON.stringify(error), status: -1 };
+      }
+    }
+  };
+
+  // direct axios fetch
+  const postDelegateMigrationOwnershipClaims = async (delegateId: string) => {
+    try {
+      const { status } = await axios.post<void>(
+        getMigrationOwnershipClaimsUrl(delegateId)
+      );
+      return { status };
+    } catch (error) {
+      return { status: -1 };
+    }
+  };
+
+  const fetchDelegateStatusPairs = () => {
+    const pairs: Array<DelegateStatusPair> = [];
+    migrationDelegatesData?.delegates?.forEach(async delegate => {
+      const result = await getDelegateMigrationStatus(delegate.sourceId);
+      if (result.status === 200) {
+        pairs.push({
+          delegate,
+          data: result.data
+        });
+      }
+      setMigrationStatusList([...pairs]);
+    });
+  };
+
+  // claim owneship for selected delegates
+  const claimOwnership = async (delegates: string[]) => {
+    try {
+      setImportInProgress(true);
+      await Promise.all(delegates.map(postDelegateMigrationOwnershipClaims));
+      enqueueSnackbar(
+        buildSnackbarItem({
+          severity: "success",
+          title: t("notifications.success"),
+          message: ""
+        })
+      );
+    } catch (error) {
+      enqueueSnackbar(
+        buildSnackbarItem({
+          severity: "error",
+          title: t("notifications.exceptionError"),
+          message: JSON.stringify(error)
+        })
+      );
+    } finally {
+      setImportInProgress(false);
+      fetchMigrationLatestData();
+    }
+  };
+
+  const fetchMigrationLatestData = () => {
+    migrationItemsFetchData(
+      "getServicesMigrationStatus",
+      {},
+      MigrationItemList,
+      { notify: "errors" }
+    );
+  };
+
+  // for each delegate fetch own migration status
+  useEffect(() => {
+    fetchDelegateStatusPairs();
```

Comment:
```suggestion
    fetchDelegateStatusPairs(migrationDelegatesData);
```

make dependencies explict otherwise it gets difficult to track why `migrationDelegatesData` appears in the deps array an what other variables should.

Reply:
`migrationDelegatesData` is the result of `useFetch` custom hook and for this reason I cannot use it "explicitly", at least in this context.
**P.S.:** Since I have an upcoming release deadline, I'm adding this issue as a technical debt task that I will evaluate asap.
Thanks for suggestions :-)